mne_icalabel/gui/_label_components.py

```diff
@@ -0,0 +1,288 @@
+import platform
+from typing import Dict, List
+
+from matplotlib import pyplot as plt
+from matplotlib.backends.backend_qt5agg import FigureCanvasQTAgg
+from matplotlib.figure import Figure
+from mne.preprocessing import ICA
+from qtpy.QtCore import Qt, Slot
+from qtpy.QtWidgets import (
+    QAbstractItemView,
+    QButtonGroup,
+    QGridLayout,
+    QListWidget,
+    QMainWindow,
+    QPushButton,
+    QVBoxLayout,
+    QWidget,
+)
+
+from mne_icalabel.config import ICLABEL_LABELS_TO_MNE
+
+_CH_MENU_WIDTH = 30 if platform.system() == "Windows" else 10
+
+
+class TopomapFig(FigureCanvasQTAgg):
+    """Topographic map figure widget."""
+
+    def __init__(self, width=4, height=4, dpi=100):
+        self.fig = Figure(figsize=(width, height), dpi=dpi)
+        self.axes = self.fig.subplots()
+        self.fig.subplots_adjust(bottom=0, left=0, right=1, top=1, wspace=0, hspace=0)
+        super().__init__(self.fig)
+
+    def reset(self) -> None:
+        """Reset the topographic plot."""
+        self.axes.clear()
+
+    def redraw(self) -> None:
+        """Redraw the data."""
+        self.fig.canvas.draw()
+        self.fig.canvas.flush_events()
+
+
+class PowerSpectralDensityFig(FigureCanvasQTAgg):
+    """PSD figure widget."""
+
+    def __init__(self, width=4, height=4, dpi=100):
+        self.fig = Figure(figsize=(width, height), dpi=dpi)
+        self.axes = self.fig.subplots()
+        self.axes.axis("off")
+        super().__init__(self.fig)
+
+    def reset(self) -> None:
+        """Reset the PSD plot."""
+        self.axes.clear()
+
+    def redraw(self) -> None:
+        """Redraw the data."""
+        self.axes.set_title("")
+        self.fig.tight_layout()
+        self.fig.canvas.draw()
+        self.fig.canvas.flush_events()
+
+
+class TimeSeriesFig(FigureCanvasQTAgg):
+    """Time-series figure widget."""
+
+    def __init__(self, width=4, height=4, dpi=100):
+        self.fig = Figure(figsize=(width, height), dpi=dpi)
+        self.axes = self.fig.subplots()
+        self.fig.subplots_adjust(bottom=0, left=0, right=1, top=1, wspace=0, hspace=0)
+        # clean up excess plot text, invert
+        self.axes.set_xticks([])
+        self.axes.set_yticks([])
+        super().__init__(self.fig)
+
+    def reset(self) -> None:
+        """Reset the time-series plot."""
+        self.axes.clear()
+
+    def change_ic(self, inst, ica, idx):
+        """Change the component time-series source to plot."""
+        # TODO: change to use ica
+        # TODO: make this work and actually embed in the plot
+        fig = inst.plot(block=False, order=[idx])
+
+        self.axes.set_title("")
+        self.fig = fig
+        self.fig.tight_layout()
+        self.fig.canvas.draw()
+        self.fig.canvas.flush_events()
```

Comment:
@marsipu this somehow plots the time-series outside the widget. Do you happen to know how to make this work based on your suggestion here: https://github.com/mne-tools/mne-qt-browser/issues/117#issuecomment-1106836014?

Context: I'm trying to add the time-series plot of one ICA source at a time into one of the subplots of the GUI. I would ideally like to just build on the work you have. Rn I'm just plotting the raw inst for testing.

Reply:
I am surprised by the `axes` and `fig` used below as those are specific to the matplotlib backend(s), and I don't think we have them in the QtBrowser.